### PR TITLE
ocamlPackages.octavius: 0.2.0 -> 1.2.2

### PR DESCRIPTION
**Pull request by superherointj**
ocamlPackages.octavius: 0.2.0 -> 1.2.2

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125891` at f05058f9 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>flow</li>
    <li>libbap</li>
    <li>patdiff</li>
    <li>python38Packages.bap</li>
    <li>python39Packages.bap</li>
    <li>satysfi</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/octavius/default.nix [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
+buildDunePackage rec {
+  pname = "octavius";
+  version = "1.2.2";
 
-stdenv.mkDerivation {
-  name = "ocaml${ocaml.version}-octavius-0.2.0";
   src = fetchurl {
-    url = "https://github.com/ocaml-doc/octavius/releases/download/v0.2.0/octavius-0.2.0.tbz";
-    sha256 = "02milzzlr4xk5aymg2fjz27f528d5pyscqvld3q0dm41zcpkz5ml";
+    url = "https://github.com/ocaml-doc/octavius/archive/refs/tags/v${version}.tar.gz";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/octavius/default.nix:8]: Please use `fetchzip` or `fetchFromGitHub`.

Comment (superherointj) [reply pkgs/development/ocaml-modules/octavius/default.nix]: Done.

**Review by vbgl — CHANGES_REQUESTED** [review]
This breaks a few things (e.g., `ocaml-ng.ocamlPackages_4_07.core`).

**Comment by superherointj** [comment]
> This breaks a few things (e.g., `ocaml-ng.ocamlPackages_4_07.core`).

```
$ nix build .#ocaml-ng.ocamlPackages_4_07.core
       > Error: Version 2.8 of dune is not supported.
       > Supported versions:
       > - 0.0
       > - 1.0 to 1.11
```

I'm not familiar with how old versions work. I'm not sure what I did wrong. Still trying to understand this. Any help is appreciated.

**Comment by vbgl** [comment]
You can set `useDune2 = lib.versionAtLeast ocaml.version "4.08"` (don’t forget to take `ocaml` as argument).

**Comment by superherointj** [comment]
> You can set `useDune2 = lib.versionAtLeast ocaml.version "4.08"` (don’t forget to take `ocaml` as argument).

Thanks!

I'm quite worried about annoying you too much. There aren't that many OCaml reviewers. I hope you don't mind the PRs.

**Comment by superherointj** [comment]
Should I be using `useDune2 = lib.versionAtLeast ocaml.version "4.08"` in (mostly) every package?

**Comment by vbgl** [comment]
Thanks for asking. Your work is appreciated and welcome but it would be better if you avoided opening too many PRs at once.

**Comment by vbgl** [comment]
> Should I be using |useDune2 = lib.versionAtLeast ocaml.version "4.08"| 
> in (mostly) every package?

No.

**Comment by superherointj** [comment]
> Thanks for asking. Your work is appreciated and welcome but it would be better if you avoided opening too many PRs at once.

I will be mindful of opening PRs. But I still may have to open some more to port Esy at least. Once I do that I can use npm/opam through Esy. It might prove less of a problem then. For now, in such goal, the critical PRs for me are:
* reason-native #125427
* reason (ppxlib migration) #126066

**Pull request merged**